RFCs/0019-Disable-Appdomain-While-Running-Tests.md

```diff
@@ -0,0 +1,28 @@
+# 0019 Disable Appdomain while Running Tests
+
+## Summary
+The RFC outlines why
+
+1. Test adapters should honour disable app domain setting
+
+2. Running of tests with disable appdomain is important for resiliency 
+
+## Motivation
+In past we have seen customer hitting issue with Appdomain.Unload. There are two main issues with Appdoamin.Unload call
+
+1. Hang in Appdoamin.Unload (Tracking issue https://github.com/Microsoft/testfx/issues/225)
+
+2. AppDomain.Unload call crashes the process even if you have an exception handler in code. This is due to a CLR and OS bug which ends up ignoring the exception handler in code and terminates the process.
+
+## Proposed changes
+
+Proposed guidelines are for customers and test adapters who wants to avoid tese issues.
+
+**Adapters**
+1. Test Adapter should honour ```<DisableAppDomain>``` setting inside RunConfiguration node of runsettings. Check https://github.com/Microsoft/vstest-docs/blob/master/docs/configure.md for information on this setting. This will ensure that adapters dont create AppDomain at all to run tests
```

Comment:
What happens if they don't honour it? I mean, there are several adapters out there, and it may be a while before they all react to this proposed change - what happens in the meantime? Can I assume that they continue to work.

Reply:
yes they will continue to work